File: intervals/252_meeting_rooms.go
import "sort"

func meetingRooms(intervals [][]int) bool {
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})

	prev := intervals[0]
	for i := 1; i < len(intervals); i++ {
		current := intervals[i]
		if prev[1] > current[0] {
			return false
		}
		prev = current
	}
	return true
}

File: intervals/252_meeting_rooms.py
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end


class Solution:
    def canAttendMeetings(self, intervals: list[Interval]) -> bool:
        if not intervals:
            return True
        intervals.sort(key=lambda x: x.start)

        last = intervals[0]
        for i in range(1, len(intervals)):
            interval = intervals[i]

            if last.end > interval.start:
                return False
            last = interval

        return True
